### PR TITLE
script to fetch Clearbit data via API, merge resulting data into email-map file

### PR DESCRIPTION
**Pull request by radoslaw**

### CODE REVIEW

**Comment by radoslaw** [comment]
code was refactored to fall withinsingle digits checking by rubocop, variable names are now more explicit

File: merge_clearbit_affiliations_into_map.rb [hunk 1]
```diff
@@ -58,7 +58,7 @@ def check_for_self_employment(affiliation_suggestion)
   selfies.concat(['software engineer', 'self-employed', 'independent'])
   selfies.concat(['self employed', 'evangelist', 'enthusiast', 'self'])
   selfies.each do |selfie|
-    affiliation_suggestion = 'Self' if company_name&.include? selfie
+    affiliation_suggestion = 'independent' if company_name&.include? selfie
```
Comment (dankohn) [inline merge_clearbit_affiliations_into_map.rb]: Independent

**Comment by lukaszgryglicki** [comment]
More tools will require changing "Self" to "Independent" in addition to this one.

File: cncf-config/email-map [hunk 2]
```diff
@@ -1,3 +1,5 @@
+
+
```
Comment (lukaszgryglicki) [inline cncf-config/email-map:2]: Get rid of empty lines

File: cncf-config/email-map [hunk 3]
```diff
@@ -9,33 +11,55 @@
...
 2001xxj!gmail.com Huawei
-21521209!zju.edu.cn Self
+21521209!zju.edu.cn Independent
 21621231!zju.edu.cn Zhejiang University
-21621232!zju.edu.cn Self
+21621232!zju.edu.cn Independent
 21651061!zju.edu.cn Zhejiang University
 21651152!zju.edu.cn Zhejiang University
 21751189!zju.edu.cn Zhejiang University
+248 NotFound
```
Comment (lukaszgryglicki) [inline cncf-config/email-map:32]: `248` is not an email

File: cncf-config/email-map [hunk 4]
```diff
@@ -9,33 +11,55 @@
...
 30670611+cliffburdick!users.noreply.github.com ViaSat
 30678878+lcfang!users.noreply.github.com Huawei
 307419146!qq.com EasyStack
+30897305+hanxiaoshuai!users.noreply.github.com NotFound
 30914745+clement-buchart!users.noreply.github.com Auchan e-Commerce France
+31228480+zhaochaobo!users.noreply.github.com NotFound
+31731825+harisanker1780!users.noreply.github.com NotFound
+31899264+erikbsap!users.noreply.github.com NotFound
 32305648+n3wscott!users.noreply.github.com NotFound
+32553353+jgoldbergstackoverdrive!users.noreply.github.com NotFound
```
Comment (lukaszgryglicki) [inline cncf-config/email-map:51]: You can only mark email with "Not Found" when You manually researched that email using google, linkedin and all possible sources. Do not mark emails as NotFound if they were not found by a single automatic tool.
"NotFound" serves as a marker that email was researched in many ways and none of them found affiliation, most important: MANUALLY!

File: cncf-config/email-map [hunk 5]
```diff
@@ -9,33 +11,55 @@
...
+32735684+franzbeltran!users.noreply.github.com NotFound
 33024221!qq.com.cn ZTE
 344677472!qq.com Awcloud
+3452489+jhernandezb!users.noreply.github.com NotFound
+3502082+wtip!users.noreply.github.com NotFound
 38098369!qq.com ZTE
+382186+charlesverdad!users.noreply.github.com NotFound
+3834+dams!users.noreply.github.com NotFound
 393926893!qq.com Caicloud
+400 NotFound
```
Comment (lukaszgryglicki) [inline cncf-config/email-map:62]: 400 is not an email.
There are more like this.
Please QA Your work, especially this file must be reviewed manually every time - this MUST be correct.

File: cncf-config/email-map [hunk 6]
```diff
@@ -1146,17 +1261,21 @@ bearnard!gmail.com Yola.com
 beasurajitroy!gmail.com Walmart
 beeps!google.com Google
 beeradb!gmail.com DRUD Tech
-beethoven1110!gmail.com Self
+beethoven1110!gmail.com missing_name
```
Comment (lukaszgryglicki) [inline cncf-config/email-map:1264]: missing_name doesn't seem to be a company name

File: cncf-config/email-map [hunk 7]
```diff
@@ -1271,14 +1403,15 @@ bmelville!google.com Google
...
 bnemec!redhat.com Red Hat
-bnemec!redhat.com Self < 2013-09-09
 bnemec!us.ibm.com IBM < 2013-07-12
+bnemec!us.ibm.com Independent < 2013-09-09
 bnemec!us.ibm.com Red Hat
-bnemec!us.ibm.com Self < 2013-09-09
 bo!godaddy.com GoDaddy
 bo.wang!easystack.cn EasyStack
 bo.wang!easystack.cn IBM < 2014-12-26
+bo0merzzz!gmail.com SAP Labs Bulgaria
```
Comment (lukaszgryglicki) [inline cncf-config/email-map:1414]: Labs means Independent?

File: cncf-config/email-map [hunk 8]
```diff
@@ -2891,39 +3156,47 @@ eli.mallon!csats.com C-SATS < 2016-01-15
...
 elliot!users.noreply.github.com everydayhero
 elliot.franford!gmail.com TempAlert
+elnappo!users.noreply.github.com NotFound
 elobatocs!gmail.com Red Hat
 elson.o.rodriguez!intel.com Intel
 elson.rodriguez!gmail.com Intel
 elvis.teixeira!lsbd.ufc.br Hitachi
 elvismtt!gmail.com Hitachi
 elyscape!gmail.com Bluescape
+ema!linux.it Linux
```
Comment (lukaszgryglicki) [inline cncf-config/email-map:3183]: Hmmm... Linux?

**Review by lukaszgryglicki — CHANGES_REQUESTED** [review]
So finally:
1. I would split replace 'Self' with 'Independent' to a separate PR, because it is hard to see actual new affiliations in this PR (not absolutely necessary)
2. I would *NOT* add "notFound" entries unless You manually checked all possible sources and You gave up. Do *NOT* set "NotFound" just because some automatic tool didn't find affiliations. NotFound is to be used as a marker for emails that were researched in all possible ways and failed to be processed.
3. Manually review email-map before creating PR - few items are not emails addresses.

**Comment by radoslaw** [comment]
my branch went berserk, will create a slimmer PR with edited code

**Pull request closed without merging**